File: LeetCode/0495. Teemo Attacking/495-111021.py
"""
Hao Ren
11 October, 2021
495. Teemo Attacking
"""

class Solution(object):
    def findPoisonedDuration(self, timeSeries, duration):
        """
        :type timeSeries: List[int]
        :type duration: int
        :rtype: int
        """
        time = 0
        length = len(timeSeries)
        for i in range(length - 1):
            time += min(timeSeries[i + 1] - timeSeries[i], duration)
        return time + duration

File: LeetCode/0495. Teemo Attacking/495-111021.c
// Hao Ren
// 11 October, 2021
// 495. Teemo Attacking

int findPoisonedDuration(int *timeSeries, int timeSeriesSize, int duration) {
    int time = 0;
    for (int i = 0; i < timeSeriesSize - 1; i++) {
        if (timeSeries[i + 1] - timeSeries[i] < duration) {
            time += timeSeries[i + 1] - timeSeries[i];
        } else {
            time += duration;
        }
    }
    time += duration;
    return time;
}
